pkg/plugins/optional/helm/v2alpha/edit.go

```diff
@@ -0,0 +1,217 @@
+
+	"sigs.k8s.io/kubebuilder/v4/pkg/config"
+	"sigs.k8s.io/kubebuilder/v4/pkg/machinery"
+	"sigs.k8s.io/kubebuilder/v4/pkg/plugin"
+	"sigs.k8s.io/kubebuilder/v4/pkg/plugin/util"
+	"sigs.k8s.io/kubebuilder/v4/pkg/plugins/optional/helm/v2alpha/scaffolds"
+)
+
+const (
+	// DefaultManifestsFile is the default path for kustomize output manifests
+	DefaultManifestsFile = "dist/install.yaml"
+	// DefaultOutputDir is the default output directory for Helm charts
+	DefaultOutputDir = "dist"
+)
+
+var _ plugin.EditSubcommand = &editSubcommand{}
+
+type editSubcommand struct {
+	config        config.Config
+	force         bool
+	manifestsFile string
+	outputDir     string
+}
+
+//nolint:lll
+func (p *editSubcommand) UpdateMetadata(cliMeta plugin.CLIMetadata, subcmdMeta *plugin.SubcommandMetadata) {
+	subcmdMeta.Description = `Generate a Helm chart from your project's kustomize output.
+
+This plugin dynamically generates Helm chart templates by parsing the output of 'make build-installer' 
+(dist/install.yaml by default). The generated chart preserves all customizations made to your kustomize 
+configuration including environment variables, labels, and annotations.
+
+The chart structure mirrors your config/ directory organization for easy maintenance.`
+
+	subcmdMeta.Examples = fmt.Sprintf(`# Generate Helm chart from default manifests (dist/install.yaml) to default output (dist/)
+  %[1]s edit --plugins=%[2]s
+
+# Generate Helm chart and overwrite existing files (useful for updates)
+  %[1]s edit --plugins=%[2]s --force
+
+# Generate Helm chart from a custom manifests file
+  %[1]s edit --plugins=%[2]s --manifests=path/to/custom-install.yaml
+
+# Generate Helm chart to a custom output directory
+  %[1]s edit --plugins=%[2]s --output=charts
+
+# Generate from custom manifests to custom output directory
+  %[1]s edit --plugins=%[2]s --manifests=manifests/install.yaml --output=helm-charts
+
+# Typical workflow:
+  make build-installer  # Generate dist/install.yaml with latest changes
+  %[1]s edit --plugins=%[2]s  # Generate/update Helm chart in dist/chart/
+
+**NOTE**: The plugin preserves customizations in values.yaml, Chart.yaml, _helpers.tpl, and .helmignore
+unless --force is used. All template files are regenerated to match your current kustomize output.
+
+The generated chart structure mirrors your config/ directory:
+<output>/chart/
+├── Chart.yaml
+├── values.yaml
+├── .helmignore
+└── templates/
+    ├── _helpers.tpl
+    ├── namespace.yaml
+    ├── rbac/
+    ├── manager/
+    ├── webhook/
+    └── ...
+`, cliMeta.CommandName, plugin.KeyFor(Plugin{}))
+}
+
+func (p *editSubcommand) BindFlags(fs *pflag.FlagSet) {
+	fs.BoolVar(&p.force, "force", false, "if true, regenerates all the files")
+	fs.StringVar(&p.manifestsFile, "manifests", DefaultManifestsFile,
+		"path to the YAML file containing Kubernetes manifests from kustomize output")
+	fs.StringVar(&p.outputDir, "output", DefaultOutputDir, "output directory for the generated Helm chart")
```

Comment:
shouldn't the flag called `output-dir`?
as you suggested in my initial PR here: https://github.com/kubernetes-sigs/kubebuilder/pull/4891#discussion_r2171168191

Reply:
fixed.